app/V1Module/presenters/ReferenceExerciseSolutionsPresenter.php

```diff
@@ -116,19 +124,25 @@ public function actionCreateReferenceSolution(string $exerciseId) {
 
     $req = $this->getRequest();
     $note = $req->getPost("note");
-    $runtimeId = $req->getPost("runtime");
-
-    $criteria = Criteria::create()->where(Criteria::expr()->eq("id", $runtimeId));
-    $configsFound = $exercise->getRuntimeConfigs()->matching($criteria);
-    $numOfResults = $configsFound->count();
-    if ($numOfResults !== 1) {
-      throw new NotFoundException("Runtime config ID not specified correctly. Got ${numOfResults} matches from exercise runtimes.");
+    $runtimeId = $req->getPost("runtimeEnvironmentId");
+
+    // detect the runtime configuration
+    if ($runtimeId !== NULL) {
+      $runtimeEnvironment = $this->runtimeEnvironments->findOrThrow($runtimeId);
+      $runtimeConfiguration = $exercise->getRuntimeConfigByEnvironment($runtimeEnvironment);
+      if ($runtimeConfiguration === NULL) {
+        throw new NotFoundException("RuntimeConfiguration was not found");
```

Comment:
Sice už je to zmergovaný, ale pro ladění se to bude hodit, až to budeme dělat na klientovi.